# DP: updates from the 16 branch upto 20260918 (documentation).

LANG=C git diff --no-renames --src-prefix=a/src/ --dst-prefix=b/src/ \
	78d4ac73dd391005b895a6148cd9831e28e1208b 222f7e062b26d227fb410f87b928035f52d105bb \
	| awk '/^diff .*\.texi/ {skip=0; print; next} /^diff / {skip=1; next} skip==0' \
        | grep -v -E '^(diff|index)'

--- a/src/gcc/doc/install.texi
+++ b/src/gcc/doc/install.texi
@@ -2492,7 +2492,7 @@ Linux systems and on big endian 64-bit systems where the default cpu
 is at least power7 (i.e.@: @option{--with-cpu=power7},
 @option{--with-cpu=power8}, or @option{--with-cpu=power9} is used).
 
-If you use the @option{--with-long-double-64} configuration option,
+If you use the @option{--without-long-double-128} configuration option,
 the @option{--with-long-double-format=ibm} and
 @option{--with-long-double-format=ieee} options are ignored.
 
--- a/src/gcc/doc/md.texi
+++ b/src/gcc/doc/md.texi
@@ -7391,7 +7391,7 @@ The operation is equivalent to:
 
 @smallexample
 int c = GET_MODE_SIZE (@var{m}) / GET_MODE_SIZE (@var{n});
-for (j = 0; j < operand4 + operand5; j++)
+for (j = 0; j < operand4 - operand5; j++)
   for (i = 0; i < c; i++)
     if (operand2[j])
       operand0[i][j] = operand1[j * c + i];
@@ -7443,7 +7443,7 @@ The operation is equivalent to:
 
 @smallexample
 int c = GET_MODE_SIZE (@var{m}) / GET_MODE_SIZE (@var{n});
-for (j = 0; j < operand3 + operand4; j++)
+for (j = 0; j < operand3 - operand4; j++)
   if (operand2[j])
     for (i = 0; i < c; i++)
       operand0[j * c + i] = operand1[i][j];
@@ -7490,11 +7490,11 @@ Like @samp{gather_load@var{m}@var{n}}, but takes an extra mask operand
 (operand 7) and a bias operand (operand 8).
 
 Similar to mask_len_load the instruction loads at
-most (operand 7 + operand 8) elements from memory.
+most (operand 7 - operand 8) elements from memory.
 Bit @var{i} of the mask is set if element @var{i} of the result should
 be loaded from memory and clear if element @var{i} of the result should
 be set to element @var{i} of operand 6.
-Mask elements @var{i} with @var{i} > (operand 7 + operand 8) are ignored.
+Mask elements @var{i} with @var{i} > (operand 7 - operand 8) are ignored.
 
 @mdindex mask_len_strided_load@var{m}
 @item @samp{mask_len_strided_load@var{m}}
@@ -7505,10 +7505,10 @@ operand 3 is mask operand, operand 4 is length operand and operand 5 is bias ope
 The instruction can be seen as a special case of @code{mask_len_gather_load@var{m}@var{n}}
 with an offset vector that is a @code{vec_series} with zero as base and operand 2 as step.
 For each element the load address is operand 1 + @var{i} * operand 2.
-Similar to mask_len_load, the instruction loads at most (operand 4 + operand 5) elements from memory.
+Similar to mask_len_load, the instruction loads at most (operand 4 - operand 5) elements from memory.
 Element @var{i} of the mask (operand 3) is set if element @var{i} of the result should
 be loaded from memory and clear if element @var{i} of the result should be zero.
-Mask elements @var{i} with @var{i} > (operand 4 + operand 5) are ignored.
+Mask elements @var{i} with @var{i} > (operand 4 - operand 5) are ignored.
 
 @mdindex scatter_store@var{m}@var{n}
 @item @samp{scatter_store@var{m}@var{n}}
@@ -7543,9 +7543,9 @@ of the result should be stored to memory.
 @item @samp{mask_len_scatter_store@var{m}@var{n}}
 Like @samp{scatter_store@var{m}@var{n}}, but takes an extra mask operand (operand 5),
 a len operand (operand 6) as well as a bias operand (operand 7).  The instruction stores
-at most (operand 6 + operand 7) elements of (operand 4) to memory.
+at most (operand 6 - operand 7) elements of (operand 4) to memory.
 Bit @var{i} of the mask is set if element @var{i} of (operand 4) should be stored.
-Mask elements @var{i} with @var{i} > (operand 6 + operand 7) are ignored.
+Mask elements @var{i} with @var{i} > (operand 6 - operand 7) are ignored.
 
 @mdindex mask_len_strided_store@var{m}
 @item @samp{mask_len_strided_store@var{m}}
@@ -7556,9 +7556,9 @@ operand 3 is mask operand, operand 4 is length operand and operand 5 is bias ope
 The instruction can be seen as a special case of @code{mask_len_scatter_store@var{m}@var{n}}
 with an offset vector that is a @code{vec_series} with zero as base and operand 1 as step.
 For each element the store address is operand 0 + @var{i} * operand 1.
-Similar to mask_len_store, the instruction stores at most (operand 4 + operand 5) elements of
+Similar to mask_len_store, the instruction stores at most (operand 4 - operand 5) elements of
 mask (operand 3) to memory.  Element @var{i} of the mask is set if element @var{i} of (operand 3)
-should be stored.  Mask elements @var{i} with @var{i} > (operand 4 + operand 5) are ignored.
+should be stored.  Mask elements @var{i} with @var{i} > (operand 4 - operand 5) are ignored.
 
 @mdindex while_ult@var{m}@var{n}
 @item @code{while_ult@var{m}@var{n}}
@@ -7768,7 +7768,7 @@ This pattern is not allowed to @code{FAIL}.
 
 @mdindex len_load_@var{m}
 @item @samp{len_load_@var{m}}
-Load (operand 3 + operand 4) elements from memory operand 1
+Load (operand 3 - operand 4) elements from memory operand 1
 into vector register operand 0.  Operands 0 and 1 have mode @var{m},
 which must be a vector mode.  Operand 3 has whichever integer mode the
 target prefers.  Operand 2 (the ``else value'') is of mode @var{m} and
@@ -7781,7 +7781,7 @@ constant bias: it is either a constant 0 or a constant -1.  The predicate on
 operand 4 must only accept the bias values that the target actually supports.
 GCC handles a bias of 0 more efficiently than a bias of -1.
 
-If (operand 3 + operand 4) exceeds the number of elements in mode
+If (operand 3 - operand 4) exceeds the number of elements in mode
 @var{m}, the behavior is undefined.
 
 If the target prefers the length to be measured in bytes rather than
@@ -7792,7 +7792,7 @@ This pattern is not allowed to @code{FAIL}.
 
 @mdindex len_store_@var{m}
 @item @samp{len_store_@var{m}}
-Store (operand 2 + operand 3) vector elements from vector register operand 1
+Store (operand 2 - operand 3) vector elements from vector register operand 1
 into memory operand 0, leaving the other elements of
 operand 0 unchanged.  Operands 0 and 1 have mode @var{m}, which must be
 a vector mode.  Operand 2 has whichever integer mode the target prefers.
@@ -7803,7 +7803,7 @@ constant bias: it is either a constant 0 or a constant -1.  The predicate on
 operand 3 must only accept the bias values that the target actually supports.
 GCC handles a bias of 0 more efficiently than a bias of -1.
 
-If (operand 2 + operand 3) exceeds the number of elements in mode
+If (operand 2 - operand 3) exceeds the number of elements in mode
 @var{m}, the behavior is undefined.
 
 If the target prefers the length to be measured in bytes
@@ -7815,23 +7815,23 @@ This pattern is not allowed to @code{FAIL}.
 @mdindex mask_len_load@var{m}@var{n}
 @item @samp{mask_len_load@var{m}@var{n}}
 Perform a masked load from the memory location pointed to by operand 1
-into register operand 0.  (operand 3 + operand 4) elements are loaded from
+into register operand 0.  (operand 4 - operand 5) elements are loaded from
 memory and other elements in operand 0 are set to undefined values.
 This is a combination of len_load and maskload.
 Operands 0 and 1 have mode @var{m}, which must be a vector mode.  Operand 3
 has whichever integer mode the target prefers.  A mask is specified in
 operand 2 which must be of type @var{n}.  The mask has lower precedence than
 the length and is itself subject to length masking,
-i.e. only mask indices < (operand 4 + operand 5) are used.
+i.e. only mask indices < (operand 4 - operand 5) are used.
 Operand 3 is an else operand similar to the one in @code{maskload}.
-Operand 4 conceptually has mode @code{QI}.
+Operand 5 conceptually has mode @code{QI}.
 
 Operand 4 can be a variable or a constant amount.  Operand 5 specifies a
 constant bias: it is either a constant 0 or a constant -1.  The predicate on
 operand 5 must only accept the bias values that the target actually supports.
 GCC handles a bias of 0 more efficiently than a bias of -1.
 
-If (operand 4 + operand 5) exceeds the number of elements in mode
+If (operand 4 - operand 5) exceeds the number of elements in mode
 @var{m}, the behavior is undefined.
 
 If the target prefers the length to be measured in bytes
@@ -7843,13 +7843,13 @@ This pattern is not allowed to @code{FAIL}.
 @mdindex mask_len_store@var{m}@var{n}
 @item @samp{mask_len_store@var{m}@var{n}}
 Perform a masked store from vector register operand 1 into memory operand 0.
-(operand 3 + operand 4) elements are stored to memory
+(operand 3 - operand 4) elements are stored to memory
 and leave the other elements of operand 0 unchanged.
 This is a combination of len_store and maskstore.
 Operands 0 and 1 have mode @var{m}, which must be a vector mode.  Operand 3 has whichever
 integer mode the target prefers.  A mask is specified in operand 2 which must be
 of type @var{n}.  The mask has lower precedence than the length and is itself subject to
-length masking, i.e. only mask indices < (operand 3 + operand 4) are used.
+length masking, i.e. only mask indices < (operand 3 - operand 4) are used.
 Operand 4 conceptually has mode @code{QI}.
 
 Operand 2 can be a variable or a constant amount.  Operand 3 specifies a
@@ -7857,7 +7857,7 @@ constant bias: it is either a constant 0 or a constant -1.  The predicate on
 operand 4 must only accept the bias values that the target actually supports.
 GCC handles a bias of 0 more efficiently than a bias of -1.
 
-If (operand 2 + operand 4) exceeds the number of elements in mode
+If (operand 3 - operand 4) exceeds the number of elements in mode
 @var{m}, the behavior is undefined.
 
 If the target prefers the length to be measured in bytes
@@ -8604,13 +8604,13 @@ for (i = 0; i < GET_MODE_NUNITS (@var{m}); i++)
 @itemx @samp{cond_len_floor@var{mode}}
 @itemx @samp{cond_len_round@var{mode}}
 @itemx @samp{cond_len_rint@var{mode}}
-When operand 1 is true and element index < operand 4 + operand 5, perform an operation on operands 1 and
+When operand 1 is true and element index < operand 4 - operand 5, perform an operation on operands 1 and
 store the result in operand 0, otherwise store operand 2 in operand 0.
 The operation only works for the operands are vectors.
 
 @smallexample
 for (i = 0; i < GET_MODE_NUNITS (@var{m}); i++)
-  op0[i] = (i < ops[4] + ops[5] && op1[i]
+  op0[i] = (i < ops[4] - ops[5] && op1[i]
             ? @var{op} op2[i]
             : op3[i]);
 @end smallexample
@@ -8670,13 +8670,13 @@ form of @samp{@var{op}@var{mode}2}.
 @itemx @samp{cond_len_ashl@var{mode}}
 @itemx @samp{cond_len_ashr@var{mode}}
 @itemx @samp{cond_len_lshr@var{mode}}
-When operand 1 is true and element index < operand 5 + operand 6, perform an operation on operands 2 and 3 and
+When operand 1 is true and element index < operand 5 - operand 6, perform an operation on operands 2 and 3 and
 store the result in operand 0, otherwise store operand 4 in operand 0.
 The operation only works for the operands are vectors.
 
 @smallexample
 for (i = 0; i < GET_MODE_NUNITS (@var{m}); i++)
-  op0[i] = (i < ops[5] + ops[6] && op1[i]
+  op0[i] = (i < ops[5] - ops[6] && op1[i]
             ? op2[i] @var{op} op3[i]
             : op4[i]);
 @end smallexample
@@ -8713,7 +8713,7 @@ takes 3 operands rather than two.  For example, the vector form of
 
 @smallexample
 for (i = 0; i < GET_MODE_NUNITS (@var{m}); i++)
-  op0[i] = (i < ops[6] + ops[7] && op1[i]
+  op0[i] = (i < ops[6] - ops[7] && op1[i]
             ? fma (op2[i], op3[i], op4[i])
             : op5[i]);
 @end smallexample
--- a/src/gcc/doc/riscv-mcpu.texi
+++ b/src/gcc/doc/riscv-mcpu.texi
@@ -64,6 +64,8 @@ by particular CPU name. Permissible values for this option are:
 
 @samp{tt-ascalon-d8},
 
+@samp{tt-ascalon-xg},
+
 @samp{xiangshan-nanhu},
 
 @samp{xiangshan-kunminghu},
--- a/src/gcc/doc/riscv-mtune.texi
+++ b/src/gcc/doc/riscv-mtune.texi
@@ -32,6 +32,8 @@ particular CPU name.  Permissible values for this option are:
 
 @samp{tt-ascalon-x},
 
+@samp{tt-ascalon-xg},
+
 @samp{tt-ascalon-d8},
 
 @samp{thead-c906},
